### PR TITLE
feat(openvpn) refactor to allow hieradata network setup

### PR DESCRIPTION
**Pull request by dduportal**
This PR refactors the OpenVPN profile to allow specifying the network configurations (routes, etc.) through hieradata, setting the ground for multiple VPN machines.

It also introduces the following changes that were required to test and validate the reafctoring:

- Add a second NIC to the container (require using Ubuntu 20.04 instead of 18.04 but EOL is near so it's OK)
- Split hieradata configs between rspec and vagrant to ensure we can properly unit tests
- Track the puppet firewall module with updatecli and bump it to 4.0.0 (to support latest iptables versions)

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -13,6 +13,9 @@ Vagrant.configure("2") do |config|
         d.has_ssh = true
     end
 
+    # Add a secondary NIC (simulation of a "private" network for roles such as openvpn)
```
Comment (lemeurherve) [inline Vagrantfile]:
```suggestion
    # Add a secondary NIC ("private" network simulation for roles such as openvpn)
```

File: dist/profile/manifests/openvpn.pp [hunk 2]
```diff
@@ -79,79 +89,77 @@
     content => template("${module_name}/openvpn/90-network-config.yaml.erb"),
   }
 
-  ## Custom routes for peered networks
-  exec { 'addroute-10.240.0.0':
-    command => 'ip route add 10.240.0.0/14 via 10.0.2.1 dev eth1',
-    unless  => 'route | grep 10.240.0.0',
-    path    => '/usr/bin:/usr/sbin:/bin:/sbin',
+  # The CLI '/sbin/route' is required to create custom routes for peered networks
+  package { 'net-tools':
```
Comment (lemeurherve) [inline dist/profile/manifests/openvpn.pp]:
```suggestion
  # The CLI '/sbin/route' included in net-tools is required to create custom routes for peered networks
  package { 'net-tools':
```

File: dist/profile/manifests/openvpn.pp [hunk 3]
```diff
@@ -79,79 +89,77 @@
...
+    ensure => present,
   }
 
+  # Allow openvpn clients (incoming 443)
   firewall { '107 accept incoming 443 connections':
```
Comment (lemeurherve) [inline dist/profile/manifests/openvpn.pp:98]: Out of curiosity, any particular reason for 10**7** here and below?

File: hieradata/rspec/profile_jenkinscontroller.yaml [hunk 4]
```diff
@@ -0,0 +1,405 @@
+---
```
Comment (lemeurherve) [inline hieradata/rspec/profile_jenkinscontroller.yaml:1]: Is it the current Jenkins controller values content?

File: updatecli/weekly.d/puppet-modules/firewall.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    scmid: default
+
+pullrequests:
+  default:
+    kind: github
+    scmid: default
+    title: Bump the Firewall Puppet Module to {{ source "latestVersion" }}
+    spec:
+      labels:
+        - puppet-module
```
Comment (lemeurherve) [inline updatecli/weekly.d/puppet-modules/firewall.yaml:57]:
```suggestion
        - puppet-module
        - puppetlabs-firewall
```

File: updatecli/weekly.d/puppet-modules/firewall.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+---
+name: Bump the Firewall Puppet Module
```
Comment (lemeurherve) [inline updatecli/weekly.d/puppet-modules/firewall.yaml]:
```suggestion
name: Bump the puppetlabs-firewall module
```
Nit, more no-brainer
A matter of taste, feel free do decline.

File: updatecli/weekly.d/puppet-modules/firewall.yaml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        kind: semver
+    transformers:
+      - trimprefix: v
+
+conditions:
+  testPuppetModuleExists:
+    kind: shell
+    disablesourceinput: true
+    spec:
+    # https://forge.puppet.com/v3/files/puppetlabs-firewall-4.0.0.tar.gz
```
Comment (lemeurherve) [inline updatecli/weekly.d/puppet-modules/firewall.yaml]:
```suggestion
```
No value added? The curl command is explicit enough IMO.

File: updatecli/weekly.d/puppet-modules/firewall.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        mod 'puppetlabs-firewall'(.*)
+      replacepattern: >
+        mod 'puppetlabs-firewall', '{{ source "latestVersion" }}'
+    scmid: default
+
+pullrequests:
+  default:
+    kind: github
+    scmid: default
+    title: Bump the Firewall Puppet Module to {{ source "latestVersion" }}
```
Comment (lemeurherve) [inline updatecli/weekly.d/puppet-modules/firewall.yaml]:
```suggestion
    name: "Update Puppetfile with the latest puppetlabs-firewall module version"
    kind: file
    sourceid: latestVersion
    spec:
      file: Puppetfile
      matchpattern: >
        mod 'puppetlabs-firewall'(.*)
      replacepattern: >
        mod 'puppetlabs-firewall', '{{ source "latestVersion" }}'
    scmid: default

pullrequests:
  default:
    kind: github
    scmid: default
    title: Bump `puppetlabs-firewall` module to {{ source "latestVersion" }}
```

**Review by lemeurherve — COMMENTED** [review]
First light review, will come back to it

File: updatecli/weekly.d/puppet-modules/firewall.yaml (hunk 7 above)
Comment (dduportal) [reply updatecli/weekly.d/puppet-modules/firewall.yaml]: leftover that I forgot to remove 👍

**Comment by dduportal** [comment]
Merging to evaluate the changes on the current machine:

- Puppet Agent stopped and disabled on vpn.jenkins.io
- A report of the noop will be added here. If problematic, then a rollback will happen.

**Comment by dduportal** [comment]
The Noop shows the following changes:

- A lot of changes in `/opt/puppetlabs/puppet/cache/` due to the puupetlabs-firewall module being updated. Expected.

- Diff related to the changes in the profile:

- `Notice: /Stage[main]/Profile::Openvpn/Firewall[107 accept incoming 443 connections]/dport: current_value , should be 443 (noop)` 
=> set destination port, should remove puppet warnings and enforce firewall rule

- `Notice: /Stage[main]/Profile::Openvpn/Firewall[107 accept incoming 443 connections]/iniface: current_value , should be 'eth0' (noop)`
=> No more VPN connections available from private netowrks to this machine (enforced security)

- `Notice: /Stage[main]/Profile::Openvpn/Firewall[107 accept incoming 22 connections]/ensure: current_value 'absent', should be 'present' (noop)`
=> No more VPN connections available from private netowrks to this machine (enforced security)

- Routing rules (masquerade) will all change:

````
Notice: /Stage[main]/Profile::Openvpn/Firewall[100 allow routing from 10.8.0.0/24 to 10.0.99.0/24 on ports 80/443]/ensure: current_value 'absent', should be 'present' (noop)
Notice: /Stage[main]/Profile::Openvpn/Firewall[100 allow routing from 10.8.1.0/24 to 10.0.99.0/24 on ports 80/443]/ensure: current_value 'absent', should be 'present' (noop)
Notice: /Stage[main]/Profile::Openvpn/Firewall[100 allow routing from 10.8.0.0/24 to 10.0.2.0/24 on ports 80/443]/ensure: current_value 'absent', should be 'present' (noop)
Notice: /Stage[main]/Profile::Openvpn/Firewall[100 allow routing from 10.8.0.0/24 to 10.240.0.0/14 on ports 80/443]/ensure: current_value 'absent', should be 'present' (noop)
Notice: /Stage[main]/Profile::Openvpn/Firewall[100 allow routing from 10.8.1.0/24 to 10.0.2.0/24 on ports 80/443]/ensure: current_value 'absent', should be 'present' (noop)
Notice: /Stage[main]/Profile::Openvpn/Firewall[100 allow routing from 10.8.1.0/24 to 10.240.0.0/14 on ports 80/443]/ensure: current_value 'absent', should be 'present' (noop)
Notice: /Stage[main]/Profile::Openvpn/Firewall[100 allow routing from 10.8.0.0/24 to 10.0.1.0/24 on ports 80/443]/ensure: current_value 'absent', should be 'present' (noop)
Notice: /Stage[main]/Profile::Openvpn/Firewall[100 allow routing from 10.8.1.0/24 to 10.0.1.0/24 on ports 80/443]/ensure: current_value 'absent', should be 'present' (noop)
```

**Comment by dduportal** [comment]
Applied with success

**Pull request merged**